api/v1alpha1/nodepool_types.go

```diff
@@ -571,6 +571,18 @@ type KubevirtCompute struct {
 	// +optional
 	// +kubebuilder:default=2
 	Cores *uint32 `json:"cores"`
+
+	// QosClass If set to "Guaranteed", requests the scheduler to place the VirtualMachineInstance on a node with
+	// enough dedicated pCPUs and pin the vCPUs to it. See more info here:
+	// https://kubevirt.io/user-guide/virtual_machines/dedicated_cpu_resources/
+	//
+	// In addition, set the VMI limit memory and CPU to be the requested values, to set the VMI as a Guaranteed QoS Class;
+	// See here for more details:
+	// https://kubevirt.io/user-guide/operations/node_overcommit/#requesting-the-right-qos-class-for-virtualmachineinstances
+	//
+	// +optional
+	//+kubebuilder:validation:Enum=Guaranteed
+	QosClass *string `json:"qosClass,omitempty"`
```

Comment:
I think we're mixing two concepts here which are separate even though they are somewhat related to each other.

The concept of "Dedicated Resources" is what the guaranteed qos gives us.  This means if we ask for 2 cpus and 10Gi of memory, the system is reserving usage of those resources for our workload when qos=guaranteed. **This is what we want right now.**

The concept of "CPU Pinning" is what the kubevirt VMI `dedicatedCpuPlacement` is involved with. We require guaranteed QOS when performing cpu pinning, but cpu pinning is not necessary for us to have dedicated resource allocations via the pod's guaranteed QOS. This might be something we want to expose to hypershift/kubevirt in the future, but it's not necessary right now.

Part of the reason this topic is convoluted is that the terminology we're using in kubevirt around "dedicated cpu resources" seems to imply that is the setting we need to guarantee dedicated resources for a VM... but really that option is related to cpu pinning in conjuction to guaranteed CPU qos.

Reply:
Thanks @davidvossel. A few questions:
1. Do we want to wait for the work done now regarding resources in KubeVirt?
2. Until now we used the memory parameter/default to populate the guest.memory field. Do we want to use *the same value* for the resources map (request/limit)? This question also related to the first question, because I don't want to override what done in KubeVirt, if it improves the VMI performance and stability.

Reply:
> Do we want to wait for the work done now regarding resources in KubeVirt?

what work in KubeVirt are you referring to?